File: spring-web-java/src/main/java/spring/web/java/infrastructure/config/security/CorsProperties.java
package spring.web.java.infrastructure.config.security;

import java.util.List;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@Component
@ConfigurationProperties(prefix = "cors")
public class CorsProperties {
	private List<String> allowedOrigins;
	private List<String> allowedMethods;
	private List<String> allowedHeaders;
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/config/security/CorsProperties.kt
package spring.web.kotlin.infrastructure.config.security

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "cors")
class CorsProperties {
    var allowedOrigins: List<String> = emptyList()
    var allowedMethods: List<String> = emptyList()
    var allowedHeaders: List<String> = emptyList()
}